code/obj/ToSplit/barber_shop.dm

```diff
@@ -437,3 +436,555 @@
+			boutput(user, "<span class='alert'>[M] doesn't have enough facial hair!</span>")
+			return 0
+
+		if((new_style in beards) && (M.bioHolder.mobAppearance.customization_second in mustaches))
+			boutput(user, "<span class='alert'>[M] doesn't have a beard!</span>")
+			return 0
+
+		if((new_style in mustaches) && (M.bioHolder.mobAppearance.customization_second in beards))
+			boutput(user, "<span class='alert'>[M] doesn't have a mustache!</span>")
+			return 0
+
+	var/our_barbery_conditions = get_barbery_conditions(M, user)
+	var/degree_of_success = 0 // 0 - 3, 0 being failure, 3 being catastrophic hair success
+	if(prob(our_barbery_conditions))
+		degree_of_success = 3
+	else // oh no we fucked up!
+		degree_of_success = clamp(0,round((our_barbery_conditions / 30) - 1), 2)
+	//and then just jam all the vars into the action bar and let it handle the rest!
+	actions.start(new/datum/action/bar/shave(M, user, degree_of_success, new_style), user)
+	return 1
+
+/obj/item/proc/do_haircut(mob/living/carbon/human/M as mob, mob/living/carbon/human/user as mob)
+	if(!M || !user)
+		return 0 // Who's cutting whose hair, now?
+
+	if(user.zone_sel.selecting != "head")
+		boutput(user, "You poke [M] with your [src]. You'll need to aim for [M]'s head to cut their hair. Or change your grip if you want to cut anything else.")
+		return 0
+
+	var/mob/living/carbon/human/H = M
+	if(ishuman(M) && ((H.head && H.head.c_flags & COVERSEYES) || (H.wear_mask && H.wear_mask.c_flags & COVERSEYES) || (H.glasses && H.glasses.c_flags & COVERSEYES)))
+		// you can't stab someone in the eyes wearing a mask!
+		boutput(user, "<span class='notice'>You're going to need to remove that mask/helmet/glasses first.</span>")
+		return 0
+
+	if(M.bioHolder.mobAppearance.customization_first == "None")
+		boutput(user, "<span class='alert'>There is nothing to cut!</span>")
+		return 0
+
+	if(!mutant_barber_fluff(M, user, "haircut"))
+		return 0
+
+	var/new_style = input(user, "Please select style", "Style")  as null|anything in customization_styles + customization_styles_gimmick
+
+	if (new_style)
+		if(M.bioHolder.mobAppearance.customization_first == "Balding" && new_style != "None")
+			boutput(user, "<span class='alert'>Not enough hair!</span>")
+			return 0
+
+	if(!new_style)
+		return 0
+
+	var/our_barbery_conditions = get_barbery_conditions(M, user)
+	var/degree_of_success = 0 // 0 - 3, 0 being failure, 3 being catastrophic hair success
+	if(prob(our_barbery_conditions))
+		degree_of_success = 3
+	else // oh no we fucked up!
+		degree_of_success = (our_barbery_conditions % 3) - 1
+	//and then just jam all the vars into the action bar and let it handle the rest!
+	actions.start(new/datum/action/bar/haircut(M, user, degree_of_success, new_style), user)
+	return 1
+
+/proc/get_barbery_conditions(mob/living/carbon/human/M as mob, mob/living/carbon/human/user as mob)
+	var/barbery_conditions = 0
+	// let's see how ideal the haircutting conditions are
```

Comment:
Mixing addition, subtration and multiplication like this is not very intuitive and I pity the person that's gonna explain it on the wiki. I recommend axing `*=`s.

Reply:
True, and it does lead to some weird numbers. I suppose I could just make the multipliers just flat addition and subtraction.